sweetest/src/main/java/com/mysugr/sweetest/framework/dependency/DependencyStates.kt

```diff
@@ -25,8 +26,7 @@ class DependencyStates(private val initializerContext: DependencyInitializerCont
 
     override val all get() = statesMap.values
 
-    override operator fun <T : Any> get(configuration: DependencyConfiguration<T>):
-        DependencyState<T> {
+    override operator fun <T : Any> get(configuration: DependencyConfiguration<T>): DependencyState<T> {
```

Comment:
instead of overriding the index operator i we should have functions `getByConfiguration` and `getByDependencyType` instead of the index operator, i think. as it is now, it's hard to understand what's happening from the caller's side...